portfolio/src/main/java/com/google/sps/servlets/ListCommentsServlet.java

```diff
@@ -32,27 +32,51 @@
 
 @WebServlet("/list-comments")
 public class ListCommentsServlet extends HttpServlet {
+  // by default show five comments
+  private int commentLimit = 5;
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
     Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
 
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery results = datastore.prepare(query);
 
     List<Comment> comments = new ArrayList<>();
+
+    int addedSoFar = 0;
     for (Entity entity : results.asIterable()) {
+      if (addedSoFar == commentLimit) {
+          return;
```

Comment:
This should be `break;` rather than `return;`. If you return here, the response isn't populated, so no comments are displayed at all once the limit is reached.